### PR TITLE
Update Fused Documentation

### PR DESCRIPTION
**Pull request by edwinleeTT**
### Ticket
#25634 

### Problem description
Docs needed an update to clarify support and provide TTNN examples

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/tg-quick-trigger.yaml) CI passes (if applicable)
- [ ] [TG demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/tg-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/normalization/groupnorm/device/groupnorm_op.cpp [hunk 1]
```diff
@@ -31,6 +31,9 @@ void GroupNorm::validate(
     TT_FATAL(a.padded_shape()[3] % this->num_groups == 0, "channel must be divisible by num_groups!");
     TT_FATAL(a.padded_shape()[1] == 1, "input tensor shape[1] must be 1!");
 
+    TT_FATAL(gamma.has_value(), "ERROR - gamma must be provided as argument to group_norm");
+    TT_FATAL(beta.has_value(), "ERROR - beta must be provided as argument to group_norm");
```
Comment (bbradelTT) [inline ttnn/cpp/ttnn/operations/normalization/groupnorm/device/groupnorm_op.cpp]: This is because in groupnorm.cpp there is:
```
    const std::optional<ttnn::Tensor>& gamma =
        weight.has_value() ? std::optional<ttnn::Tensor>(ttnn::unsqueeze_to_4D(weight.value())) : std::nullopt;
    const std::optional<ttnn::Tensor>& beta =
        bias.has_value() ? std::optional<ttnn::Tensor>(ttnn::unsqueeze_to_4D(bias.value())) : std::nullopt;
```
right?

The if statements below should be removed as well.
```
if (gamma.has_value()) {
```
and
```
if (beta.has_value()) {
```

File: ttnn/cpp/ttnn/operations/normalization/batch_norm/batch_norm_pybind.cpp [hunk 2]
```diff
@@ -57,18 +61,31 @@ void bind_batch_norm_operation(py::module& module) {
...
+            weight = ttnn.rand([1, C, 1, 1], dtype=ttnn.DataType.BFLOAT16, layout=ttnn.TILE_LAYOUT, device=device)
+            bias = ttnn.from_torch(torch.rand([1, C, 1, 1], dtype=torch.bfloat16), layout=ttnn.TILE_LAYOUT, device=device)
+
+            output = ttnn.batch_norm(
+                input_tensor,
+                running_mean = running_mean,
+                running_var = running_var,
+                weight = weight,
+                bias = bias,
+                eps =  1e-05,
```
Comment (bbradelTT) [inline ttnn/cpp/ttnn/operations/normalization/batch_norm/batch_norm_pybind.cpp]: One less space after the equal sign.

File: ttnn/cpp/ttnn/operations/normalization/groupnorm/groupnorm.cpp [hunk 3]
```diff
@@ -48,8 +48,7 @@ ttnn::Tensor ExecuteGroupNorm::invoke(
 
     TT_FATAL(
         input_tensor.memory_config().memory_layout() != TensorMemoryLayout::WIDTH_SHARDED,
-        "Unsupported memory layout: Input tensor must be width-sharded, but it is not. (memory_layout={})",
-        input_tensor.memory_config().memory_layout());
+        "Unsupported memory layout: Input tensor cannot be width-sharded.");
```
Comment (bbradelTT) [inline ttnn/cpp/ttnn/operations/normalization/groupnorm/groupnorm.cpp:51]: This fatal should probably be left as is. Good to know what the incorrect config was.

**Review by bbradelTT — APPROVED** [review]
Please also include the mathematical formula for softmax if it's not there already.

File: ttnn/cpp/ttnn/operations/normalization/groupnorm/device/groupnorm_op.cpp (hunk 1 above)
Comment (edwinleeTT) [reply ttnn/cpp/ttnn/operations/normalization/groupnorm/device/groupnorm_op.cpp]: My understanding is that these were intended to be optional, but the support was never completed. Trying to run an example on my end resulted in a segfault, which is why I added these assertions until the support is ready. It seems SDXL is using group norm without supplying them though, which is a surprise to us. As far as I can tell, it is fully functioning in their use case.

I'm going to remove these assertions for now and this should be explored with #26594

File: ttnn/cpp/ttnn/operations/normalization/groupnorm/groupnorm.cpp (hunk 3 above)
Comment (edwinleeTT) [reply ttnn/cpp/ttnn/operations/normalization/groupnorm/groupnorm.cpp:51]: I think the original text was reversed from the assertion. i.e., the problem here is when the input IS width-sharded, whereas the original message was that it must be width-sharded instead of {something else}.

Original assertion message is along the lines of "input tensor must be width-sharded, but it's width sharded"
